### PR TITLE
json: added custom marsheler

### PR DESCRIPTION
**Pull request by hunchback**
seems to be minimally working:

```
go test -v
=== RUN   TestUuid
--- PASS: TestUuid (0.00s)
=== RUN   TestOptionsLength
expected: ["map",[["mac_prefix","0e:f6:a4"],["max_tunid","16711680"],["northd_internal_version","20.12.0-20.14.0-52.0"],["northd_probe_interval","5000"],["svc_monitor_mac","e2:d2:ac:ad:7b:8a"],["e2e_timestamp","1612966696"]]]
actual  : ["map",[["e2e_timestamp","1612966696"],["mac_prefix","0e:f6:a4"],["max_tunid","16711680"],["northd_internal_version","20.12.0-20.14.0-52.0"],["northd_probe_interval","5000"],["svc_monitor_mac","e2:d2:ac:ad:7b:8a"]]]
--- PASS: TestOptionsLength (0.00s)
=== RUN   TestLength
expected: {"_uuid":["uuid","a5088a51-7756-4dd4-909c-b7c59c9fcce7"],"name":"","nb_cfg":0,"nb_cfg_timestamp":0,"sb_cfg":0,"sb_cfg_timestamp":0,"hv_cfg":0,"hv_cfg_timestamp":0,"external_ids":["map",[]],"connections":["uuid","71a565df-188f-42d0-9f18-74e180e27889"],"ssl":["set",[]],"options":["map",[["mac_prefix","0e:f6:a4"],["max_tunid","16711680"],["northd_internal_version","20.12.0-20.14.0-52.0"],["northd_probe_interval","5000"],["svc_monitor_mac","e2:d2:ac:ad:7b:8a"],["e2e_timestamp","1612966696"]]],"ipsec":false}
actual  : {"_uuid":["uuid","a5088a51-7756-4dd4-909c-b7c59c9fcce7"],"name":"","nb_cfg":0,"nb_cfg_timestamp":0,"sb_cfg":0,"sb_cfg_timestamp":0,"hv_cfg":0,"hv_cfg_timestamp":0,"external_ids":["map",[]],"connections":["uuid","71a565df-188f-42d0-9f18-74e180e27889"],"ssl":["set",[]],"options":["map",[["e2e_timestamp","1612966696"],["mac_prefix","0e:f6:a4"],["max_tunid","16711680"],["northd_internal_version","20.12.0-20.14.0-52.0"],["northd_probe_interval","5000"],["svc_monitor_mac","e2:d2:ac:ad:7b:8a"]]],"ipsec":false}
--- PASS: TestLength (0.00s)
PASS
ok      github.com/roytman/ovsdb-etcd/pkg/json  0.003s
```

please review

### CODE REVIEW

File: pkg/json/encode.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	"fmt"
+)
+
+type Uuid string
+
+type Map map[string]string
+
+type Set []string
+
+type NB_Global struct {
```
Comment (roytman) [inline pkg/json/encode.go]: NB_Global is not part of the encoder

File: pkg/json/encode.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	ExternalIds     Map      `json:"external_ids"`
+	Connections      Uuid     `json:"connections"`
+	Ssl              Set `json:"ssl"`
+	Options          Map      `json:"options"`
+	Ipsec            bool     `json:"ipsec"`
+}
+
+func (u Uuid) MarshalJSON() ([]byte, error) {
+	return json.Marshal([]string{"uuid", string(u)})
+}
```
Comment (roytman) [inline pkg/json/encode.go:11]: Please add additional type - <named-uuid>
      
> A 2-element JSON array that represents the UUID of a row inserted
>       in an "insert" operation within the same transaction.  The first
>       element of the array must be the string "named-uuid", and the
>       second element should be the <id> specified as the "uuid-name" for
>       an "insert" operation within the same transaction.  For example,
>       if an "insert" operation within this transaction specifies a
>       "uuid-name" of "myrow", the following <named-uuid> represents the
>       UUID created by that operation:
> 
>       ["named-uuid", "myrow"]

https://tools.ietf.org/html/rfc7047#section-5

File: pkg/json/encode.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	"bytes"
+	"encoding/json"
+	"fmt"
+)
+
+type Uuid string
+
+type Map map[string]string
+
+type Set []string
```
Comment (roytman) [inline pkg/json/encode.go]: in addition to strings, Set can contain other types,

File: pkg/json/encode_test.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	"svc_monitor_mac":         "e2:d2:ac:ad:7b:8a",
+}
+
+var nb NB_Global = NB_Global{
+	Uuid:        uuid,
+	Connections: "71a565df-188f-42d0-9f18-74e180e27889",
+	Options:     options,
+}
+
+func TestUuid(t *testing.T) {
```
Comment (roytman) [inline pkg/json/encode_test.go:84]: For this kind of tests, Ginkgo `Table Driven Tests` are more suitable

File: pkg/json/encode_test.go [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+	b, _ := json.Marshal(uuid)
+	actual := string(b)
+	assert.Equal(t, expectedUuid, actual, "they should be equal")
+}
+
+func TestOptionsLength(t *testing.T) {
```
Comment (roytman) [inline pkg/json/encode_test.go:96]: You don't have unit test for Set
The Set parsing is wrong, it should be `["set",["clustered","standalone"]]`, we have  `["set",[["clustered"],["standalone"]]]`

File: pkg/json/encode_test.go [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+	b, _ := json.Marshal(options)
+	actual := string(b)
+	fmt.Printf("expected: %s\n", expectedOptions)
```
Comment (roytman) [inline pkg/json/encode_test.go]: pls remove the `Printfs`

**Review by roytman — CHANGES_REQUESTED** [review]

File: pkg/json/encode_test.go (hunk 4 above)
Comment (hunchback) [reply pkg/json/encode_test.go:84]: I know Ginkgo, but this is more a matter to taste .. and I am not crazy about that style of testing

File: pkg/json/encode_test.go (hunk 5 above)
Comment (hunchback) [reply pkg/json/encode_test.go:96]: ok

File: pkg/json/encode_test.go (hunk 6 above)
Comment (hunchback) [reply pkg/json/encode_test.go]: ok

File: pkg/json/encode.go (hunk 3 above)
Comment (hunchback) [reply pkg/json/encode.go]: ok

File: pkg/json/encode.go (hunk 1 above)
Comment (hunchback) [reply pkg/json/encode.go]: ok

File: pkg/json/encode.go (hunk 2 above)
Comment (hunchback) [reply pkg/json/encode.go:11]: ok

**Comment by hunchback** [comment]
@roytman - I addressed outstanding comments - please review

File: pkg/json/encode_test.go (hunk 4 above)
Comment (roytman) [reply pkg/json/encode_test.go:84]: ovn-kubernetes uses Ginko, I prefer to be in-line with them, we will "sell" our code to this team

File: pkg/json/encode.go (hunk 2 above)
Comment (roytman) [reply pkg/json/encode.go:11]: the first string should be "named-uuid", you set "myuuid"

File: pkg/json/encode_test.go (hunk 5 above)
Comment (roytman) [reply pkg/json/encode_test.go:96]: You did not fix  the set encoding

File: pkg/json/encode_test.go (hunk 4 above)
Comment (hunchback) [reply pkg/json/encode_test.go:84]: ok

File: pkg/json/encode_test.go (hunk 5 above)
Comment (hunchback) [reply pkg/json/encode_test.go:96]: ahhh ... I see will do

File: pkg/json/encode.go (hunk 2 above)
Comment (hunchback) [reply pkg/json/encode.go:11]: ok

File: pkg/json/encode_test.go (hunk 4 above)
Comment (hunchback) [reply pkg/json/encode_test.go:84]: I will not have time to handle now, my preference is that this be delayed for later and that the PR is merged without handling this

**Pull request merged**